cpp/oneapi/dal/algo/knn/backend/cpu/infer_kernel_kd_tree.cpp

```diff
@@ -40,7 +39,7 @@ using daal_knn_kd_tree_kernel_t = daal_knn::prediction::internal::
 
 template <typename Float, typename Task>
 static infer_result<Task> call_daal_kernel(const context_cpu &ctx,
-                                           const descriptor_t &desc,
+                                           const detail::descriptor_base<Task> &desc,
```

Comment:
descriptor_t has been removed because we added support of search task for both methods. We can find a similar situation in SVM for multiclass kernel support.

https://github.com/oneapi-src/oneDAL/blob/22d95a6f0d04545fd26f024e8ddfa26b1d47bb87/cpp/oneapi/dal/algo/svm/backend/cpu/infer_cls_kernel.cpp#L55-L62